src/input_handler.cpp

```diff
@@ -0,0 +1,52 @@
+#include "input_handler.h"
+
+namespace arch {
+    InputHandler InputHandler::input_handler_instance;
+    std::unordered_map<int, std::function<void()>> InputHandler::binded_table;
+    GLFWwindow *InputHandler::_window {};
+    
+    InputHandler::InputHandler() {}
+
+    InputHandler& InputHandler::get() {
+        return input_handler_instance;
+    }
+
+    void InputHandler::set_current_window(GLFWwindow* window) {
```

Comment:
This function is named `set_current_window` but, in reality, does much more. You could do something I am doing with other systems. Create init function that does all initialization and that will be explicitly run in engine.cpp.